adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -1391,5 +1391,5 @@ func getDefaultResourceMethods(apiType string) []string {
 
 func getDefaultVersionBasepath(basePath string, version string) string {
 	context := strings.ReplaceAll(basePath, "/"+version, "")
-	return fmt.Sprintf("(%s|%s)", basePath, context)
+	return fmt.Sprintf("(?:%s|%s)", basePath, context)
```

Comment:
Can you please explain the purpose behind `?`

Reply:
This is to make "(%s|%s)" behave as a non-capturing group in the regex. Behavior we saw where the basepath got placed again in the locations of the path parameters, gets resolved when we add this. I'll add a comment here.

Reply:
Great. Thanks for the explaination.

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/2878/commits/c9b1c53fe6698d6413e3057d3915dd31808bf1df